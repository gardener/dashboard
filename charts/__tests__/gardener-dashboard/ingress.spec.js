#!/usr/bin/env node
//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { basename } = require('path')
const { helm, helper } = fixtures
const { getPrivateKey, getCertificate } = helper

const chart = basename(__dirname)
function renderTemplates (templates, values) {
  return helm.renderChartTemplates(chart, templates, values)
}

describe('gardener-dashboard', function () {
  describe('ingress', function () {
    const name = 'gardener-dashboard-ingress'
    const tlsSecretName = 'other-gardener-dashboard-tls'

    let templates

    beforeEach(() => {
      templates = [
        'ingress',
        'secret-tls'
      ]
    })

    it('should render the template with tls and a secret', async function () {
      const values = {
        ingress: {
          tls: {
            secretName: tlsSecretName,
            key: getPrivateKey('tls.key'),
            crt: getCertificate('tls.crt')
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(2)
      const [ingress, tlsSecret] = documents
      expect(ingress).toMatchSnapshot()
      expect(tlsSecret).toMatchSnapshot()
    })

    it('should render the template with tls', async function () {
      const values = {
        ingress: {
          tls: {
            secretName: tlsSecretName
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(2)
      const [ingress, tlsSecret] = documents
      expect(ingress.metadata.name).toBe(name)

      expect(ingress.spec.tls).toHaveLength(1)
      expect(ingress.spec.tls[0]).toEqual({
        secretName: tlsSecretName,
        hosts: ['gardener.garden.example.org']
      })

      expect(tlsSecret).toBeFalsy()
    })

    it('should render the template without tls', async function () {
      const values = {
        ingress: {
          tls: null
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(2)
      const [ingress, tlsSecret] = documents
      expect(ingress.metadata.name).toBe(name)

      expect(ingress.spec.tls).toBeUndefined()

      expect(tlsSecret).toBeFalsy()
    })
  })
})
