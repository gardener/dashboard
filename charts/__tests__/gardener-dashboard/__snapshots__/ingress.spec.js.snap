// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard ingress should render the template with tls and a secret 1`] = `
Object {
  "apiVersion": "extensions/v1beta1",
  "kind": "Ingress",
  "metadata": Object {
    "annotations": Object {
      "kubernetes.io/ingress.class": "nginx",
      "nginx.ingress.kubernetes.io/ssl-redirect": "true",
      "nginx.ingress.kubernetes.io/use-port-in-redirects": "true",
    },
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "gardener-dashboard-0.2.0",
      "heritage": "Helm",
      "release": "dashboard",
    },
    "name": "gardener-dashboard-ingress",
    "namespace": "garden",
  },
  "spec": Object {
    "rules": Array [
      Object {
        "host": "gardener.garden.example.org",
        "http": Object {
          "paths": Array [
            Object {
              "backend": Object {
                "serviceName": "gardener-dashboard-service",
                "servicePort": 8080,
              },
              "path": "/",
              "pathType": "Prefix",
            },
          ],
        },
      },
    ],
    "tls": Array [
      Object {
        "hosts": Array [
          "gardener.garden.example.org",
        ],
        "secretName": "other-gardener-dashboard-tls",
      },
    ],
  },
}
`;

exports[`gardener-dashboard ingress should render the template with tls and a secret 2`] = `
Object {
  "apiVersion": "v1",
  "data": Object {
    "tls.crt": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmRHeHpMbU55ZEE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t",
    "tls.key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpkR3h6TG10bGVRPT0KLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0=",
  },
  "kind": "Secret",
  "metadata": Object {
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "gardener-dashboard-0.2.0",
      "heritage": "Helm",
      "release": "dashboard",
    },
    "name": "other-gardener-dashboard-tls",
    "namespace": "garden",
  },
  "type": "kubernetes.io/tls",
}
`;
